File: Windows Phone 8.1/2.5.7.0/Decompilation/Vine.WindowsPhone/Vine.Models.Analytics/ClientEvent.cs
using System;
using Newtonsoft.Json;

namespace Vine.Models.Analytics;

public abstract class ClientEvent
{
	[JsonProperty("app_state")]
	public AppState AppState { get; set; }

	[JsonProperty("device_data")]
	public DeviceData DeviceData { get; set; }

	[JsonProperty("navigation")]
	public AppNavigation Navigation { get; set; }

	[JsonProperty("client_id")]
	public string ClientId { get; set; }

	[JsonProperty("event_type")]
	public string EventType { get; set; }

	[JsonProperty("event_details")]
	public EventDetails EventDetails { get; set; }

	protected ClientEvent(string eventType)
	{
		try
		{
			EventType = eventType;
			AppState = AppState.GetDefault();
			DeviceData = DeviceData.GetDefault();
			ClientId = "windows/" + ApplicationSettings.Current.ClientVersion;
		}
		catch (Exception)
		{
		}
	}
}

File: Android/Vine-5.50.1/Decompilation/src/java/co/vine/android/scribe/model/ClientEvent.java
package co.vine.android.scribe.model;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

@JsonObject
/* loaded from: classes.dex */
public class ClientEvent {

    @JsonField(name = {"app_state"})
    public ApplicationState appState;

    @JsonField(name = {"client_id"})
    public String clientId;

    @JsonField(name = {"device_data"})
    public DeviceData deviceData;

    @JsonField(name = {"event_details"})
    public EventDetails eventDetails;

    @JsonField(name = {"event_type"})
    public String eventType;

    @JsonField(name = {"navigation"})
    public AppNavigation navigation;
}
